Feat: Add consumer group examples to ACL plugin docs.

Guaris opened the pull request:
### Description

For: https://github.com/Kong/kong-ee/pull/7603

- [x] Needs copyedit
~- [x] Needs more examples (@jschmid1)~

### Testing

Preview: https://deploy-preview-6773--kongdocs.netlify.app/hub/kong-inc/acl/dev/how-to/consumer-groups/

### Checklist 

- [x] Review label added <!-- (see below) -->
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)


Jira; https://konghq.atlassian.net/browse/KAG-3167

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
```diff
@@ -0,0 +1,92 @@
+title: ACL Examples
+---
+
+## ACL Configuration with Kong Gateway
```
```suggestion
## ACL configuration with {{site.base_gateway}}
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
```diff
@@ -0,0 +1,92 @@
+
+## ACL Configuration with Kong Gateway
+
+In this example, we will cover a common use case: As an API Owner, you want to regulate access based on the type of request methods and consumer groups. Specifically, our goal is to allow consumers in the dev group to perform GET, POST, and PUT requests on all routes, while reserving the DELETE request functionality exclusively for consumers in the admin group.
```
```suggestion
This example covers a common use case: as an API owner, you want to regulate access based on the type of request methods and consumer groups. Specifically, the goal is to allow consumers in the dev group to perform GET, POST, and PUT requests on all routes, while reserving the DELETE request functionality exclusively for consumers in the admin group.
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
```diff
@@ -0,0 +1,92 @@
+
+### Create consumer groups 
+
+Using the API, create a consumer group for  `dev`
```
Can you put these into steps (ordered list)?

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
````diff
@@ -0,0 +1,92 @@
+  --data '{"name":"admin",}'
+``` 
+
+Add a consumer to the `admin` group. Then populate those consumer groups with consumers using the UUID of the specific consumer:
````
```suggestion
Add a consumer to the `admin` group by using the UUID of the specific consumer:
```
This is what's actually happening in this step.

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
````diff
@@ -0,0 +1,92 @@
+    --data '{"name":"dev"}'
+```
+
+Then create a consumer group for `admin`
````
```suggestion
Then create a consumer group for `admin`:
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
```diff
@@ -0,0 +1,92 @@
+    --url http://localhost:8001/consumer_groups \
+    --header 'Content-Type: application/json' \
+    --header 'accept: application/json' \
+    --data '{"name":"dev"}'
```
Why are we passing attributes as json data for these examples? Normally we use the default content type and send each attribute separately. I can understand if the goal is to provide konnect instructions, but then the URL makes no sense.

I think we should either convert these examples to the standard format we use in the docs (eg `--data name=dev`), or have two tabs for each section: one for Gateway, and one for Konnect. Then this formatting makes sense.

Guaris replied:
@jschmid1

jschmid1 replied:
No specific reason, I'll update this PR with the `--data "key=value"` formatting.

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
````diff
@@ -0,0 +1,92 @@
+  --data '{"consumer":"8a4bba3c-7f82-45f0-8121-ed4d2847c4a4"}'
+```
+
+### Create Routes 
````
```suggestion
### Create routes 
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
```diff
@@ -0,0 +1,92 @@
+
+### Create Routes 
+
+Using the Admin API and the [expressions router](/gateway/latest/key-concepts/routes/expressions/) create two routes, one that matches `GET`, `POST` and `PUT`, and one that only matches `DELETE`. 
```
```suggestion
Using the Admin API and the [expressions router](/gateway/latest/key-concepts/routes/expressions/), create two routes: one that matches `GET`, `POST` and `PUT`, and one that only matches `DELETE`. 
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
```diff
@@ -0,0 +1,92 @@
+
+Using the Admin API and the [expressions router](/gateway/latest/key-concepts/routes/expressions/) create two routes, one that matches `GET`, `POST` and `PUT`, and one that only matches `DELETE`. 
+
+A route that matches when the method is _not_ `DELETE`
```
```suggestion
Create a route that matches when the method is _not_ `DELETE`:
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
````diff
@@ -0,0 +1,92 @@
+  --form-string expression='http.path == "/example" && http.method != "DELETE"'
+```
+
+A route that matches when the method _is_ DELETE
````
```suggestion
Create a route that matches when the method _is_ DELETE:
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
````diff
@@ -0,0 +1,92 @@
+```
+
+
+### Setup the ACL plugin
````
```suggestion
### Set up the ACL plugin
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
```diff
@@ -0,0 +1,92 @@
+
+### Setup the ACL plugin
+
+Scope the plugin to each of these routes with the respective `allow` configuration
```
```suggestion
Scope the plugin to each of these routes with the respective `allow` configuration.
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
```diff
@@ -0,0 +1,92 @@
+
+Scope the plugin to each of these routes with the respective `allow` configuration
+
+The `devs-and-admin` route will have the `allow` field set to both of the groups
```
```suggestion
Enable the ACL plugin on the `devs-and-admin` route, setting the `allow` field to accept both groups:
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
````diff
@@ -0,0 +1,92 @@
+```bash
+curl -X POST http://localhost:8001/routes/devs-and-admin/plugins \
+    --data "name=acl"  \
+    --data "config.allow=dev"  \
+    --data "config.allow=admin" \
````
I don't think this will work. I'm pretty sure curl needs `[]` for an array, eg `--data config.allow[]=dev` (here's an example: https://docs.konghq.com/hub/kong-inc/acl/how-to/#associate-consumers-with-an-acl)

```suggestion
    --data "config.allow[]=dev"  \
    --data "config.allow[]=admin"
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
Can we rename this file to "_consumer-groups.md"

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
```diff
@@ -0,0 +1,92 @@
+---
+nav_title: ACL Examples
+title: ACL Examples
```
```suggestion
nav_title: Using ACLs with consumer groups
title: Using ACLs with consumer groups
```
"ACL examples" is super generic and tells me nothing.

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
````diff
@@ -0,0 +1,92 @@
+    --data "config.allow=admin" \
+```
+
+The `only-admins` route will have the `allow` field set to only the `admin` group.
````
```suggestion
Enable another ACL plugin instance on the `only-admins` route, setting the `allow` field set to only accept the `admin` group:
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_examples.md:
````diff
@@ -0,0 +1,92 @@
+```bash
+curl -X POST http://localhost:8001/routes/admins-only/plugins \
+    --data "name=acl"  \
+    --data "config.allow=admin" \
````
```suggestion
    --data "config.allow=admin"
```

lena-larionova commented on app/_hub/kong-inc/acl/overview/_index.md:
```diff
@@ -3,10 +3,13 @@ nav_title: Overview
+(such as [Basic Authentication](/hub/kong-inc/basic-auth/), [Key Authentication](/hub/kong-inc/key-auth/),
+[OAuth 2.0](/hub/kong-inc/oauth2/) or [OpenID Connect](/hub/kong-inc/openid-connect/)) to have been already
 enabled on the service or route.
+You can also enable the usage of `consumer-groups` by setting the config option `include_consumer_groups` to `true`.
```
```suggestion

You can also enable the usage of consumer groups by setting the config option [`include_consumer_groups`](/hub/kong-inc/acl/configuration/#include_consumer_groups) to `true`.
```

lena-larionova commented on app/_hub/kong-inc/acl/overview/_index.md:
```diff
@@ -3,10 +3,13 @@ nav_title: Overview
+[OAuth 2.0](/hub/kong-inc/oauth2/) or [OpenID Connect](/hub/kong-inc/openid-connect/)) to have been already
 enabled on the service or route.
+You can also enable the usage of `consumer-groups` by setting the config option `include_consumer_groups` to `true`.
+This option will not only take the `ACL groups` but also the (Consumer Groups(link to page)) into consideration when evaluating
+the `allow/deny` fields.
```
```suggestion
This option lets {{site.base_gateway}} take both ACL groups and consumer groups into consideration when evaluating the `allow` and `deny` fields.
```

lena-larionova commented on app/_hub/kong-inc/acl/overview/_index.md:
```diff
@@ -16,3 +19,4 @@ You can't configure an ACL with both `allow` and `deny` configurations. An ACL w
 * [Basic configuration example](/hub/kong-inc/acl/how-to/basic-example/)
 * [Learn how to use the plugin](/hub/kong-inc/acl/how-to/)
 * [ACME plugin API reference](/hub/kong-inc/acl/api/)
+* [Examples](/hub/kong-inc/acl/how-to/examples/)
```
```suggestion
* [Using ACLs with consumer groups](/hub/kong-inc/acl/how-to/consumer-groups/)
```

Guaris commented:
@lena-larionova I renamed the file and added an ordered list. @jschmid1 could you respond to Lena's comment here:
https://github.com/Kong/docs.konghq.com/pull/6773#discussion_r1456500272

lena-larionova commented:
As of yesterday, we're no longer using version branches for development. I just reset this branch based on the new release approach. You can now work directly off of `main` and we'll be using conditional tags + a dev version.

This means that version tagging is now very, very important. Please make sure to use `if_version` tags where appropriate so we don't accidentally publish content that shouldn't be in the latest released version.

Guaris commented:
Updated to the new versioning schema

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_consumer-groups.md.md:
````diff
@@ -0,0 +1,93 @@
+       --data '{"name":"admin",}'
+     ```
+
+1. Add a consumer to the `admin` group by using the UUID of the specific consumer:
````
```suggestion
1. Using the UUID of an existing consumer, add a consumer to the `admin` group:
```
Wait - you need to have a consumer before you do this? This step doesn't _create_ a consumer, correct? Only adds an existing one? This wasn't mentioned anywhere in prerequisites. Can go one of two ways with this: add steps to create, then add a consumer, or just say you need to have consumers available in a prerequisites section.

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_consumer-groups.md.md:
````diff
@@ -0,0 +1,93 @@
+1. Add a consumer to the `admin` group by using the UUID of the specific consumer:
+
+    ```bash
+    curl --request POST \
+      --url http://localhost:8001/consumer_groups/admin/consumers \
+      --header 'Content-Type: application/json' \
+      --header 'accept: application/json' \
+      --data '{"consumer":"8a4bba3c-7f82-45f0-8121-ed4d2847c4a4"}'
````
```suggestion
    curl -i -X POST http://localhost:8001/consumer_groups/admin/consumers \
      --data "consumer=8a4bba3c-7f82-45f0-8121-ed4d2847c4a4"
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_consumer-groups.md.md:
````diff
@@ -0,0 +1,93 @@
+      --header 'accept: application/json' \
+      --data '{"consumer":"8a4bba3c-7f82-45f0-8121-ed4d2847c4a4"}'
+    ```
+1. Then add a consumer to the `dev` group. 
````
```suggestion
1. Add a different consumer to the `dev` group:
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_consumer-groups.md.md:
````diff
@@ -0,0 +1,93 @@
+1. Then add a consumer to the `dev` group. 
+
+    ```bash
+    curl --request POST \
+      --url http://localhost:8001/consumer_groups/dev/consumers \
+      --header 'Content-Type: application/json' \
+      --header 'accept: application/json' \
+      --data '{"consumer":"8a4bba3c-7f82-45f0-8121-ed4d2847c4a4"}'
````
```suggestion
    curl -i -X POST http://localhost:8001/consumer_groups/dev/consumers \
      --data "consumer=8a4bba3c-7f82-45f0-8121-ed4d2847c4a4"
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_consumer-groups.md.md:
````diff
@@ -0,0 +1,93 @@
+
+Scope the plugin to each of these routes with the respective `allow` configuration.
+
+1. Enable the ACL plugin on the `devs-and-admin` route, setting the `allow` field to accept both groups:
+
+    ```bash
+    curl -X POST http://localhost:8001/routes/devs-and-admin/plugins \
+        --data "name=acl"  \
+        --data "config.allow[]=dev"  \
+        --data "config.allow[]=admin"
+    ```
````
```suggestion
Enable the ACL plugin on the `devs-and-admin` route, setting the `allow` field to accept both groups:

<!--vale off-->
{% plugin_example %}
plugin: kong-inc/acl
name: acl
config:
  allow:
    - dev
    - admin
targets:
  - route
formats:
  - curl
  - konnect
  - yaml
  - kubernetes
{% endplugin_example %}
<!--vale on-->
```
Turning this into an example that'll generate four different formats.

Only downside is that the example doesn't have the route name pre-filled. We'll work on adding support for that (currently on the to-do list).

jschmid1 replied:
That's way better! Thanks for picking that up. We'll need to add `include_consumer_groups: true` to the examples when some of the groups are "cosumer groups" rather than acl groups.

lena-larionova replied:
Added the config to the suggested example 👍

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_consumer-groups.md.md:
````diff
@@ -0,0 +1,93 @@
+        --data "config.allow[]=admin"
+    ```
+
+1. Enable another ACL plugin instance on the `only-admins` route, setting the `allow` field set to only accept the `admin` group:
+
+    ```bash
+    curl -X POST http://localhost:8001/routes/admins-only/plugins \
+        --data "name=acl"  \
+        --data "config.allow=admin"
+    ```
````
```suggestion
Enable another ACL plugin instance on the `only-admins` route, setting the `allow` field set to only accept the `admin` group:

<!--vale off-->
{% plugin_example %}
plugin: kong-inc/acl
name: acl
config:
  allow:
    - admin
targets:
  - route
formats:
  - curl
  - konnect
  - yaml
  - kubernetes
{% endplugin_example %}
<!--vale on-->
```

jschmid1 replied:
same here -> https://github.com/Kong/docs.konghq.com/pull/6773#discussion_r1471127508

lena-larionova replied:
adjusted the config here too

lena-larionova commented:
Versioning looks good: 
3.6: https://deploy-preview-6773--kongdocs.netlify.app/hub/kong-inc/acl/dev/how-to/consumer-groups/
vs
3.5: https://deploy-preview-6773--kongdocs.netlify.app/hub/kong-inc/acl/ - no consumer groups topic in how-to section

lena-larionova commented on app/_hub/kong-inc/acl/overview/_index.md:
```diff
@@ -3,11 +3,14 @@ nav_title: Overview
+[OAuth 2.0](/hub/kong-inc/oauth2/) or [OpenID Connect](/hub/kong-inc/openid-connect/)) to have been already
 enabled on the service or route.
 
+You can also enable the usage of consumer groups by setting the config option [`include_consumer_groups`](/hub/kong-inc/acl/configuration/#include_consumer_groups) to `true`.
+This option lets {{site.base_gateway}} take both ACL groups and consumer groups into consideration when evaluating the `allow` and `deny` fields.
+
```
```suggestion
{% if_plugin_version gte:3.6.x %}
You can also enable the usage of consumer groups by setting the config option [`include_consumer_groups`](/hub/kong-inc/acl/configuration/#include_consumer_groups) to `true`.
This option lets {{site.base_gateway}} take both ACL groups and consumer groups into consideration when evaluating the `allow` and `deny` fields.
{% endif_plugin_version %}
```

lena-larionova commented on app/_hub/kong-inc/acl/how-to/_consumer-groups.md.md:
````diff
@@ -0,0 +1,124 @@
+        --header 'accept: application/json' \
+        --data '{"username":"dev"}'
+    ```
+The response will contain a UUID in the `id` field that you will need for the rest of the guide. 
````
```suggestion
The response for each request contains a UUID in the `id` field that you will need for the rest of the guide. 
```
